tests/test_pasqal.py

```diff
@@ -13,13 +13,116 @@
 # limitations under the License.
 from __future__ import annotations
 
+import dataclasses
 from pathlib import Path
+from typing import Any
+from unittest.mock import patch
+
+import pytest
 
 import pulser
 import pulser_pasqal
+from pulser.devices import Chadoq2
+from pulser.register import Register
+from pulser.sequence import Sequence
+from pulser_pasqal import Configuration, DeviceType, Endpoints, PasqalCloud
 
 root = Path(__file__).parent.parent
 
 
 def test_version():
     assert pulser_pasqal.__version__ == pulser.__version__
+
+
+@dataclasses.dataclass
+class CloudFixture:
+    pasqal_cloud: PasqalCloud
+    mock_cloud_sdk: Any
+
+
+@pytest.fixture
+def fixt():
+    with patch("sdk.SDK", autospec=True) as mock_cloud_sdk_class:
+        pasqal_cloud_kwargs = dict(
+            client_id="abc",
+            client_secret="def",
+            endpoints=Endpoints(core="core_url", account="account_url"),
+            webhook="xyz",
+        )
+
+        pasqal_cloud = PasqalCloud(**pasqal_cloud_kwargs)
+
+        mock_cloud_sdk_class.assert_called_once_with(**pasqal_cloud_kwargs)
+
+        mock_cloud_sdk = mock_cloud_sdk_class.return_value
+
+        yield CloudFixture(
+            pasqal_cloud=pasqal_cloud, mock_cloud_sdk=mock_cloud_sdk
+        )
```

Comment:
Nice solution! I have a question: why use `yield` instead of `return` here? Is there even a diference?